src/controllers/eventController.ts

```diff
@@ -1,18 +1,35 @@
 // Controller class for the Event API endpoints
 // The controller has a class and method that call the repository.
 import { pool } from "../utils";
+import { RowDataPacket } from "mysql2";
+import { parseDataToEventType } from "../types/types";
 
 class EventsController {
   async getAllEvents() {
-    const [result] = await pool.query("SELECT * FROM Event");
-    return result;
+    const [data] = await pool.query("SELECT * FROM Event");
+
+    const parsedRowData = data as RowDataPacket[];
+    const Events = parsedRowData
```

Comment:
Is this a sort of type syntax we want to go with, basically after casting to a RowDataPacket, we then map each element in the list to their typed version if possible then filter out the falsy/null values in end, the main issue is that this still includes the |null in the type afterwards although it shouldn't

Reply:
If we hit a falsy/null value, shouldn't the request fail? Assuming the system is deterministic.

Reply:
For proper input, why would any values be falsy in the first place?

Reply:
Well this is mainly used to case the values to the proper types/variables, if the request fails it will just return an empty array, but this mainly used to get the correct variables and ignore any crap within the database that doesn't conform/doesn't have all the attributes